### DISCUSSION TITLE
AMD crash defender service is still running after installation of driver.

### DISCUSSION THREAD

**Discussion by usama7628674** [GSDragoon/RadeonSoftwareSlimmer#18] [General]
After modifying the driver from this tool I installed it, but AMD crash defender service is still running in services.msc even though I've unchecked the option before modifying the driver.

**Comment by GSDragoon** [comment]
There's no way to prevent the service from installing. Not without a driver repack. The only way to stop it is to disable the service after the driver is installed and make sure the Issue Detection option is disabled. You might also need the Crash Defender Manager Driver disabled too.

Under Settings -> System 
![image](https://user-images.githubusercontent.com/37422356/179427170-62091f21-e227-46b9-9b07-0962754ee455.png)

**Comment by usama7628674** [comment]
Okay, got it. I've disabled crash defender service. It's not a big deal, however, I'm pretty sure I've unchecked HDMI audio or whatever HDMI garbage is bundled with driver, but I still have HDMI playback in sound settings. Is it normal?

![image](https://user-images.githubusercontent.com/29252277/179519141-02b2267a-5982-4171-8782-2f09a68de46e.png)

**Reply by GSDragoon** [reply]
Even if you don't install the AMD audio drivers, Windows is still going to see the device and install generic drivers for it. In order to disable those devices you need to disable the High Definition Audio Device associated with them. Depending on what other audio devices you have, it may be a bit tricky to find the right one. I find viewing devices by connection to be the helpful way to tell.

From Device Manager: View -> Devices by connection
Then find the PCI Express port that has your Radeon GPU. Within that same port should be a High Definition Audio Controller. Disable the High Definition Audio Device under the audio controller with the GPU. Your devices will probably be a little different, but should look something like this.
![image](https://user-images.githubusercontent.com/37422356/180589886-fcd5fd71-ee9f-4b91-85e3-383fa2120a5c.png)

If you disable the audio controller, instead of the device, I forget exactly what happens but it might be something like new drivers that cause the audio device to show up new again. Disabling the device works better.

**Comment by usama7628674** [comment]
@GSDragoon Is there any way to disable AMD game advisor?

**Reply by GSDragoon** [reply]
Looking at my notes, you **might** be able to add a value called `GamePerformanceAdviserEnabled` in `Computer\HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Class\{4d36e968-e325-11ce-bfc1-08002be10318}\0000` with a value of 1 to disable part of it from Radeon Software. If the RadeonSoftware process isn't running, it won't collect any game data.

**Reply by usama7628674** [reply]
Seems to be working after applying registry edit.

**Reply by kalimaenergy** [reply]
Thank GSDragoon , maybe other good tweak to do ?

**Reply by GSDragoon** [reply]
I've been tempted, but they seem to change and break every so often. There have been some I was using, then they stopped working and causing Radeon Software to crash. It's still on my list to explore more, but haven't gotten around to it yet.

**Reply by usama7628674** [reply]
Radeon Software is back now and still collecting game performance data. This introduces micro to big stutters. Any other tweak to restrict it from collecting data while the Radeon software process is running? Sorry, but I have to run it to overclock my GPU.

**Comment by kalimaenergy** [comment]
Hello GSDragoon,

We can't add powershell script with your software on the installer?  For disabled HDMI Devices and force disabled of service.

**Reply by GSDragoon** [reply]
Disabling a device and getting it right, not disabling the wrong item sounds really dangerous. How would one automatically determine the correct device? Everyone's system is going to be different.

**Reply by kalimaenergy** [reply]
.INF HDMI AMD Radeon have only 6 hardware different ID, we can disable only HDMI radeon with this way. Easy with powershell. I can provide that. 

For the service , it's the name

EDIT : if you want make that just on post install it can be good think.

**Comment by mvildosolae** [comment]
How can we effectively disable/remove/never-install-at-all that hideous AMD Crash Defender thing? Even if I remove it before setup with RadeonSoftwareSlimmer, it still gets installed. If I remove the service and/or the driver, AMD still re-installs it after a reboot (it also shows that message about hardware configuration having changed).

**Comment by Decay312** [comment]
Here is a way to "disable" AMD Crash Defender:

1. Navigate to C:\Windows\System32\DriverStore\FileRepository , in there you will find a folder called something like this "amdfendr.inf_amd64_LONGNUMBER". There might be more than one folder, just use the newest one (look at the 'date modified' of the files within the folders)
 
2. Right click on the folder, click on Properties, click on Security, click on advanced, change the owner from System to your windows account/user (you can type the beginning of your user name and click on check names to auto complete), make sure to tick the box "Replace owner on subcontainers and objects". Then press apply and ok.

3. Back in the Security tab of Properties: Click on 'Everyone', then click on 'edit...', then tick the box that says 'Full control', then press apply and ok.

4. Final step is to go into the folder and rename amdfendr.sys to amdfendr.sys.bak

**Reply by ghost** [reply]
Worked, because stopping and disabling the Crash Defender service didn’t seem to fully stop it from running.

My main issue with the crash defender was it kept writing to the disk excessively in the eeudumps folder
